src/Illuminate/Auth/Events/Authenticated.php

```diff
@@ -8,30 +8,16 @@ class Authenticated
 {
     use SerializesModels;
 
-    /**
-     * The authentication guard name.
-     *
-     * @var string
-     */
-    public $guard;
-
-    /**
-     * The authenticated user.
-     *
-     * @var \Illuminate\Contracts\Auth\Authenticatable
-     */
-    public $user;
-
     /**
      * Create a new event instance.
      *
      * @param  string  $guard
      * @param  \Illuminate\Contracts\Auth\Authenticatable  $user
      * @return void
      */
-    public function __construct($guard, $user)
-    {
-        $this->user = $user;
-        $this->guard = $guard;
+    public function __construct(
+        public $guard,
+        public $user,
+    ) {
```

Comment:
In situations like this, do we lose type hint information in IDEs? Since we have removed the `@var` blocks.

Reply:
The `__construct` doc block should be sufficient here.

Reply:
It is.

![afbeelding](https://github.com/user-attachments/assets/ef9556ab-7774-43bc-ad92-328e22216274)